### PR TITLE
Don't require libjson headers to #include Serialbox

### PR DESCRIPTION
**Pull request by andyspiros**
This change makes the life easier for Serialbox users: libjson headers are no more installed and the user does not have to put the libjson headers directory accessible to the compiler.

Now libjson is really just an internally-used library only (except the user has to link against `libjson.a` until we have a shared library available for everyone...).

I converted `#include "libjson.h"` into forward declarations of `JSONNode`.

I tested the unittests on my machines, and the python library, and they seem to work perfectly.

### CODE REVIEW

File: libs/libjson/CMakeLists.txt [hunk 1]
```diff
@@ -1,38 +1,5 @@
 # ADDING JSON
 
-set ( JSON_HEADERS
```
Comment (cosunae) [inline libs/libjson/CMakeLists.txt:3]: I think is important that all headers are added to the library as well, not only sources. 
Many IDE's like qt creator require that in order to include them in the project. 
If you find painful adding the files by hand you can do a cmake file glob

Comment (andyspiros) [inline libs/libjson/CMakeLists.txt:3]: I see. So, you propose to revert the change on `json_files`, so all headers are also included in it?

BTW: no, I don't find it painful to list all files to the CMakeLists.txt file. It might be error prone in a project which changes a lot rapidly, but this is definitely not the case.

**Comment by cosunae** [comment]
On the other hand, I think other projects like stella or dycore might use json. 
In that case, the day we remove serialization framework from stella, I would also remove json from stella and get it from serialbox instead in order to avoid multiple copies of libjson spread around...

**Comment by andyspiros** [comment]
Ok, you are right. In facts, I don't care too much whether libjson is installed or not, to me it is important that the user does not need to make sure it can be found by the compiler.

To me, we can throw away the second commit and only merge the first one.

**Comment by pspoerri** [comment]
In the short term we want to switch to serialbox anyway.

**Comment by andyspiros** [comment]
Carlos' point is important: the dycore uses libjson to save the data about performance. We need to decide whether we want the dycore to come with its own version of libjson, to use the one installed by Serialbox, or to take another decision.

**Comment by cosunae** [comment]
I would say, since libjson is not maintained anymore, and we started fixing it, it would be better to keep only the one in seriabox (Anyhow the dycore will depend on the serialbox)

the first commit is totally fine for me

**Comment by andyspiros** [comment]
Ok, I will revert the second one. Let's how this plays out in github.

**Comment by andyspiros** [comment]
Sounds good, doesn't it?

**Comment by cosunae** [comment]
perfect :+1:

**Comment by andyspiros** [comment]
I changed the name of the PR to match the new goal. Thanks!
